app/models/movie.rb

```diff
@@ -2,4 +2,10 @@ class Movie < ApplicationRecord
   has_many :actor_movies
   has_many :actors, through: :actor_movies
   belongs_to :director
+  has_many :movie_genres
+  has_many :genres, through: :movie_genres
+
+  def self.average_rating
+    average(:rating)
```

Comment:
since we only wanted the average to be called from genres, it would have better to move this to the Genre model and done like `movies.average(:rating)` so it would use the AR association.